client/components/map/LeafletMap.tsx

```diff
@@ -0,0 +1,39 @@
+// Map Component for Leaflet.js
+import { MapContainer, TileLayer } from "react-leaflet";
+import { LatLngExpression, MapOptions } from "leaflet";
+import { useState, useEffect } from "react";
+import useMapContext from "../../hooks/useMapContext";
+import { useBreakpointValue } from "@chakra-ui/react";
+import "leaflet/dist/leaflet.css";
+
+export const LeafletMap: React.FC<
+	{
+		center: LatLngExpression;
+		children: JSX.Element | JSX.Element[];
+		zoom: number;
+	} & MapOptions
+> = ({ ...options }) => {
+	const { setMap } = useMapContext();
+	const [isClient, setIsClient] = useState(false);
+
+	useEffect(() => {
+		setIsClient(true);

```

Comment:
why do we need a state that immediately is set to true?

Reply:
I think this is to ensure that the map component only loads when the component is mounted in the client side